extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -529,86 +529,51 @@ static public function onSkinSubPageSubtitleAfterTitle( $title, &$ptext, &$cssCl
 	/**
 	 * @brief Adds an action button on user talk archive page
 	 *
-	 * @param $response
-	 * @param $ns
-	 * @param $skin
-	 * @return bool
+	 * @param array $button 'Edit' button and dropdown
+	 * @return bool true
 	 *
 	 * @author Andrzej 'nAndy' Łukaszewski
 	 */
-	static public function onPageHeaderIndexAfterActionButtonPrepared( $response, $ns, $skin ) {
+	static public function onPageHeaderIndexAfterActionButtonPrepared( array &$button ) {
 		$app = F::App();
 		$helper = new WallHelper();
 
 		if ( !empty( $app->wg->EnableWallExt ) ) {
 			$title = $app->wg->Title;
+			$ns = $title->getNamespace();
 			$parts = explode( '/', $title->getText() );
-			$action = $response->getVal( 'action' );
-			$dropdown = $response->getVal( 'dropdown' );
 			$canEdit = $app->wg->User->isAllowed( 'editwallarchivedpages' );
 
-			if ( $ns === NS_USER_WALL
+			if (
+				$ns === NS_USER_WALL
 				&& $title->isSubpage()
 				&& !empty( $parts[1] )
 				&& mb_strtolower( str_replace( ' ', '_', $parts[1] ) ) === mb_strtolower( $helper->getArchiveSubPageText() )
 			) {
-				// user talk archive
```

Comment:
I can see 2 differences here:
* we check namespace on different object: https://github.com/Wikia/app/blob/ef9e673f7725829ff0f88d022ca74c8739d33367/extensions/wikia/Wall/WallHooksHelper.class.php#L580-580 - I don't know if it's for purpose or not unfortunately.
* `$userTalkPageTitle` has different value: https://github.com/Wikia/app/blob/ef9e673f7725829ff0f88d022ca74c8739d33367/extensions/wikia/Wall/WallHooksHelper.class.php#L586-586